src/compiler/moduleNameResolver.ts

```diff
@@ -705,8 +706,11 @@ export function getConditions(options: CompilerOptions, esmMode?: boolean) {
     // conditions are only used by the node16/nodenext/bundler resolvers - there's no priority order in the list,
     // it's essentially a set (priority is determined by object insertion order in the object we look at).
     const conditions = esmMode || getEmitModuleResolutionKind(options) === ModuleResolutionKind.Bundler
-        ? ["node", "import"]
-        : ["node", "require"];
+        ? ["import"]
+        : ["require"];
+    if (getEmitModuleResolutionKind(options) !== ModuleResolutionKind.Bundler) {
+        conditions.unshift("node");
+    }

```

Comment:
Someone suggested dropping `node` from the hard-coded set of conditions when `moduleResolution` is `bundler` and I think that makes sense. I didn’t do it at first because Webpack includes the `node` condition, oddly enough, but esbuild doesn’t unless its target is set to Node. It definitely makes more sense to leave it up to `customConditions`.

Reply:
Small nit suggestion: Instead of unshift can we first add "node" and then others so we arent moving array elements

Reply:
I don't think the order matters; I just didn’t want to change a bunch of baselines unnecessarily. I did `unshift` because I remember Ron saying something about if you make an empty array and then push a non-number into it, that does a deopt. Unshift is probably worse though... I think I’ll just push `node` on last and change some traces.

Reply:
Interestingly conditions can be a `Set|undefined` i think.. we always check for contains(array, key) so may be set is better.. But thats not important for this PR.. just mentioning.